src/masonite/interfaces/Interface.py

```diff
@@ -1,19 +1,40 @@
 import inspect
+import os
+
 from .exceptions import InterfaceException
 
 
+def _is_interfaces_enabled():
```

Comment:
What do you think above this being done as a protected method on the interface? This would achieve the same thing and also also people to override the behavior (by overriding the method)

Reply:
Ok cool, can do. Out of interest, what is the use-case you were thinking of for this?

Reply:
ummm .. hmm .. can't think of one. I feel like i knew when i wrote that comment. I think it adds the possibility to do so without any downsides.

The trick when building a framework / package is to take into account not only use cases but also edge cases we don't know exist yet. Adding ability to override the method accounts for pretty much any edge cases we haven't thought of. Since that allows modification at the contract level on how it decides to enable or disable itself.

Reply:
I'm running into an issue. I haven't worked with class introspection much, so I may have things wrong. I moved the function inside the class to make it a private method, and gave it a double-underscore in front of the name. As I understood it, the interface logic should not require the overriding of the method (because it checks for the ```"__"``` in ```key```, so that it is optional for the end-user. However, after having done this, all the unit tests fail; they require this new method to be overridden.
Looking into it, I see that ```key``` in this case is ```_Interface__is_interfaces_enabled```, and so it doesn't skip it, since it doesn't begin with ```"__"```. I noticed that ```method.__name__``` contains only the method's name, so should we be checking that rather than ```key```?
But as I said, I may have this completely wrong, and missing something obvious. :-P

Reply:
which line does that key exist? Maybe need to add it to this block https://github.com/MasoniteFramework/interfaces/pull/3/files#diff-f1e93ae064fa409b2bd64b3775236e6cR57